### ISSUE TITLE
Giving wrong error "Data table should have at least 1 data row"

### ISSUE THREAD

**Issue by sguptatw** [getgauge/gauge#341]
### Expected behavior

Should run the spec successfully
### Actual behavior

Data table should have at least 1 data row
### Steps to reproduce

1.

```
Table parameter hyphen not getting executed
===================================

     |something|
     |---------|
     |-        |
     |||
     ||

Should execute successfully
-------------------------------
tags: something

* Print <something>

```
1. With the implementation

```
    @Step("Print <something>")
    public void implementation12(Object arg0) {
        System.out.println(arg0);
    }
```
1. It gives an error saying 

```
Data table should have at least 1 data row
```

---
1. Another test case

```
Table parameter hyphen not getting executed
===================================

     |something|
     ||

Should execute successfully
-------------------------------
tags: something

```
### Gauge version

```
0.4.0
```

**Comment by sguptatw** [comment]
another test case updated

**Comment by apoorvam** [comment]
The first case mentioned here is working as expected. 

If a table is given as below, it considers the hyphen as a separator between table header and its rows. But not as a data table row.
```
     |name|
     | -  |
```

If hyphen is the value that should be in first row of table, it should be given as:
```
     |name|
     |----|
     |-   |
```
The markdown table format should be as:
```
     |name |
     |-----|
     |gauge|
     |gocd |
```

**Comment by sguptatw** [comment]
The scenario 
```
 |name|
     |----|
     |-   |
```

was fixed after this bug was raised and since it is being tested on the latest code now it is fixed.
However the updated scenario

```
Table parameter hyphen not getting executed
===================================

     |something|
     ||

Should execute successfully
-------------------------------
tags: something
```

is yet to be tested 

considering that 

```
Table parameter hyphen not getting executed
===================================

     |something|
     |aa|

Should execute successfully
-------------------------------
tags: something
```
executes successfully, either the above issue is to be fixed or "header separator should be made mandatory"

**Comment by apoorvam** [comment]
The specific case where table separator is not present and first row of table is empty is also fixed.

```
Table parameter hyphen not getting executed
===================================

     |something|
     ||

Should execute successfully
-------------------------------
tags: something
```

Fix should be available in nightly >= 17.1.2017
